src/pages/settings/tabs/ApiKeys.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState } from 'react';
+import { mdiClipboardTextOutline } from '@mdi/js';
+import cx from 'classnames';
+import { sortBy, uniqueId } from 'lodash';
+import { useCopyToClipboard } from 'usehooks-ts';
+
+import Button from '@/components/Input/Button';
+import Input from '@/components/Input/Input';
+import toast from '@/components/Toast';
+import { useCreateApiToken, useDeleteApiToken } from '@/core/react-query/auth/mutations';
+import { useApiKeyQuery } from '@/core/react-query/auth/queries';
+import useEventCallback from '@/hooks/useEventCallback';
+
+import type { AuthToken } from '@/core/types/api/authToken';
+
+const UserApiTokens = (prop: { userInfo: AuthToken, onDeleteToken?: () => void }) => {
+  const { onDeleteToken, userInfo: { Device, Username } } = prop;
+  const { mutate: deleteToken } = useDeleteApiToken();
+  const onDeleteClick = useEventCallback(() => {
+    deleteToken(Device);
+    toast.success('API Key Deleted', `API Key ${Device} has been deleted!`, {
+      autoClose: 3000,
+    });
+    onDeleteToken?.();
+  });
+
+  return (
+    <div className="flex flex-row justify-between">
+      <div className="flex w-[65%] flex-col gap-y-1">
+        <span className="font-light">{Username}</span>
+        <span className="font-normal">{Device}</span>
+      </div>
+      <Button buttonType="danger" className="px-4 py-2" onClick={onDeleteClick}>
+        Delete API Key
+      </Button>
+    </div>
+  );
+};
+
+const ApiKeys = () => {
+  const [keyValue, setKeyValue] = useState('');
+  const [isRefetch, setIsRefetch] = useState(false); // eslint-disable-line @typescript-eslint/no-unused-vars
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [, copy] = useCopyToClipboard();
+  const onKeyChange = useEventCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    setKeyValue(event.target.value);
+  });
+
+  const {
+    data: createdToken,
+    isError: generatedFailed,
+    isSuccess: generatedSucceed,
+    mutate: createApiToken,
+  } = useCreateApiToken();
+
+  const handleCopyToClipboard = useEventCallback(() => {
+    if (!generatedSucceed) return;
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    copy(keyValue).then((isCopied) => {
+      if (isCopied) {
+        toast.success('Copied', 'API Key has been copied to clipboard!', {
+          autoClose: 3000,
+          toastId: 'api-copied',
+        });
+      }
+    });
+  });
+
+  const onGenerateClick = useEventCallback(() => {
+    createApiToken(keyValue);
+  });
+
+  const { data: tokens, refetch } = useApiKeyQuery();
+
+  const onTokenDelete = useEventCallback(() => {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    setIsRefetch(_ => true);
+  });
+
+  useEffect(() => {
+    if (!refetch) return;
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
```

Comment:
Same as in other place add `.catch(error.log)` no need to eslint-disable-next-line

Reply:
Why is refetch not called directly but in a useEffect?